### ISSUE TITLE
SSL_get_verify_result() is not X509_V_OK, even if verify_cb() erased error 

### ISSUE THREAD

**Issue by y-ohishi** [libressl-portable/portable#766]
I updating error to X509_V_OK in verify_cb() to ignore some errors. 

e.g. ignore X509_V_ERR_CERT_NOT_YET_VALID
```
int verify_callback(int ok, X509_STORE_CTX *ctx)
{
  if (ok == 0) {
    int error = X509_STORE_CTX_get_error(ctx);

    switch (error) {
    case X509_V_ERR_CERT_NOT_YET_VALID:
      X509_STORE_CTX_set_error(ctx, 0);
      ok = 1;
      break;
    default:
      break;
    }
  }
  return ok;
}
```

Maybe version 3.3.0 later, The content of the error updated in verify_cb() remains in SSL_get_verify_result().
I think that the problem is that the error updated in X509_STORE_CTX remains in x509_verify_ctx, so I have applied the following patch at hand.

```
--- crypto/x509/x509_verify.c
+++ crypto/x509/x509_verify.c
@@ -611,13 +611,19 @@ static int
 x509_verify_cert_error(struct x509_verify_ctx *ctx, X509 *cert, size_t depth,
     int error, int ok)
 {

        ctx->error = error;
        ctx->error_depth = depth;
        if (ctx->xsc != NULL) {
+               int ret;
+
                ctx->xsc->error = error;
                ctx->xsc->error_depth = depth;
                ctx->xsc->current_cert = cert;
-               return ctx->xsc->verify_cb(ok, ctx->xsc);
+               ret = ctx->xsc->verify_cb(ok, ctx->xsc);
+               /* override error */
+               if (ctx->error != ctx->xsc->error)
+                       ctx->error = ctx->xsc->error;
+               return ret;
        }
        return ok;
 }
```

Can you accept this changes?

**Comment by bob-beck** [comment]
Yeah this is an issue, but the override there isn't the way to solve it.  I have another candidate fix at the moment that should address this for you

**Comment by bob-beck** [comment]
Fix has been committed to current.

**Comment by bob-beck** [comment]
will leave this open until a new portable is cut

**Comment by 4a6f656c** [comment]
This should be addressed in LibreSSL 3.6.0.
